src/test/unit/PaymentCoordinatorUnit.t.sol

```diff
@@ -0,0 +1,158 @@
+        require(dummyToken.balanceOf(address(paymentCoordinator)) - balanceBefore == sum, "incorrect token balance");
+    }
+
+    function testPostMerkleRootFromUnauthorizedAddress(address unauthorizedAddress) external {
+        cheats.assume(unauthorizedAddress != paymentCoordinator.rootPublisher());
+        cheats.startPrank(unauthorizedAddress);
+        cheats.expectRevert(bytes("PaymentCoordinator: Only rootPublisher"));
+        paymentCoordinator.postMerkleRoot(bytes32(0), 0, 0);
+        cheats.stopPrank();
+    }
+
+    function testPostMerkleRoot(bytes32 root, uint256 height, uint256 calculatedUpToBlockNumber) public {
+        PaymentCoordinator.MerkleRootPost memory merkleRootPost;
+        merkleRootPost.root = root;
+        merkleRootPost.height = height;
+        merkleRootPost.calculatedUpToBlockNumber = calculatedUpToBlockNumber;
+
+        uint256 numMerkleRootsBefore = paymentCoordinator.merkleRootsLength();
+
+        cheats.startPrank(paymentCoordinator.rootPublisher());
+        paymentCoordinator.postMerkleRoot(root, height, calculatedUpToBlockNumber);
+        cheats.stopPrank();
+
+        (bytes32 rootPost, uint256 heightPost, uint256 confirmedAtBlockNumber, uint256 calculatedUpToBlockNumberPost) = paymentCoordinator.merkleRoots(numMerkleRootsBefore);
+        require(rootPost == root, "root should be set");
+        require(heightPost == height, "height should be set");
+        require(confirmedAtBlockNumber == block.number + rootPostDelay, "confirmedAtBlockNumber should be set");
+        require(calculatedUpToBlockNumberPost == calculatedUpToBlockNumber, "calculatedUpToBlockNumber should be set");
+    }
+
+    function testWithdrawEigenLayerShare(uint256[] memory amounts, address recipient) external {
+        cheats.assume(recipient != address(0));
+        testMakePayment(amounts);
+        uint256 amountToClaim = paymentCoordinator.cumulativeEigenLayerTokeEarnings(dummyToken);
+        cheats.startPrank(paymentCoordinator.rootPublisher());
+        paymentCoordinator.withdrawEigenlayerShare(dummyToken, recipient);
+        cheats.stopPrank();
+        require(paymentCoordinator.cumulativeEigenLayerTokeEarnings(dummyToken) == 0, "cumulativeEigenLayerTokeEarnings not updated correctly");
+        require(dummyToken.balanceOf(recipient) == amountToClaim, "incorrect token balance");
+
+        emit log_named_uint("balance", dummyToken.balanceOf(recipient));
+    }
+
+    function testNullifyMerkleRoot(bytes32 root, uint256 height, uint256 calculatedUpToBlockNumber) external {
+        testPostMerkleRoot(root, height, calculatedUpToBlockNumber);
+        paymentCoordinator.nullifyMerkleRoot(paymentCoordinator.merkleRootsLength() - 1);
+        (bytes32 rootPost, , , ) = paymentCoordinator.merkleRoots(paymentCoordinator.merkleRootsLength() - 1);
+        require(rootPost == bytes32(0), "root should be set");
```

Comment:
see my comment about deleting the whole struct instead of just resetting one entry. I would also update this test to ensure the whole struct is deleted (just compare the returned struct to a new, empty struct instantiation -- this may require comparing hashes of the structs).